controllers/saleorder/saleorder_storage.go

```diff
@@ -0,0 +1,108 @@
+package saleorder
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	organizationv1 "github.com/appuio/control-api/apis/organization/v1"
+	odooclient "github.com/appuio/go-odoo"
+)
+
+type Odoo16Credentials = odooclient.ClientConfig
+
+type Odoo16Options struct {
+	SalesOrderClientReference string
+}
+
+const defaultSaleOrderState = "sale"
+
+type SaleOrderStorage interface {
+	CreateSaleOrder(organizationv1.Organization) (string, error)
+	GetSaleOrderName(organizationv1.Organization) (string, error)
+}
+
+type Odoo16SaleOrderStorage struct {
```

Comment:
As above, I would really not use the Odoo version number here, just:

```suggestion
type OdooSaleOrderStorage struct {
```

This is something which might (and with 99% confidence will) be true with future Odoo versions